core/sparqlbuilder/src/test/java/org/eclipse/rdf4j/sparqlbuilder/examples/sparql11spec/Section6Test.java

```diff
@@ -61,4 +64,65 @@ public void example_6_3() {
 						GraphPatterns.and(x.has(foaf.iri("homepage"), hpage)).optional());
 		p();
 	}
+
+	@Test
+	public void example_6_4() {
+		Prefix dc = SparqlBuilder.prefix("dc", iri(DC_NS)), ns = SparqlBuilder.prefix("ns", iri(EXAMPLE_ORG_NS));
+		Variable title = SparqlBuilder.var("title"), price = SparqlBuilder.var("price"), x = SparqlBuilder.var("x");
+
+		GraphPatternNotTriples pricePattern = GraphPatterns.and(x.has(ns.iri("price"), price))
+				.filter(Expressions.or(Expressions.lt(price, 20),
+						Expressions.and(Expressions.lt(price, 50), Expressions.gt(price, 30))))
+				.optional();
+
+		query.prefix(dc, ns).select(title, price).where(x.has(dc.iri("title"), title), pricePattern);
+		Assert.assertThat(query.getQueryString(), CoreMatchers.containsString("( ?price < 50 && ?price > 30 )"));
+	}
+
+	@Test
+	public void example_6_5() {
+		Prefix dc = SparqlBuilder.prefix("dc", iri(DC_NS)), ns = SparqlBuilder.prefix("ns", iri(EXAMPLE_ORG_NS));
+		Variable title = SparqlBuilder.var("title"), price = SparqlBuilder.var("price"), x = SparqlBuilder.var("x");
+
+		GraphPatternNotTriples pricePattern = GraphPatterns.and(x.has(ns.iri("price"), price))
+				.filter(Expressions.or(Expressions.lt(price, 20),
+						Expressions.and(Expressions.gt(price, 50),
+								Expressions.or(Expressions.gt(price, 60), Expressions.lt(price, 70)))))
+				.optional();
+
+		query.prefix(dc, ns).select(title, price).where(x.has(dc.iri("title"), title), pricePattern);
+		Assert.assertThat(query.getQueryString(), CoreMatchers.containsString("( ?price < 20 || ( ?price > 50 &&" +
+				" ( ?price > 60 || ?price < 70 ) ) )"));
+	}
+
+	@Test
+	public void example_6_6() {
+		Prefix dc = SparqlBuilder.prefix("dc", iri(DC_NS)), ns = SparqlBuilder.prefix("ns", iri(EXAMPLE_ORG_NS));
+		Variable title = SparqlBuilder.var("title"), price = SparqlBuilder.var("price"), x = SparqlBuilder.var("x");
+
+		GraphPatternNotTriples pricePattern = GraphPatterns.and(x.has(ns.iri("price"), price))
+				.filter(Expressions.or(Expressions.lt(price, Expressions.subtract(Rdf.literalOf(20),
+						Expressions.multiply(Rdf.literalOf(2), Rdf.literalOf(5)))),
+						Expressions.lt(price, 50)))
+				.optional();
+
+		query.prefix(dc, ns).select(title, price).where(x.has(dc.iri("title"), title), pricePattern);
+		Assert.assertThat(query.getQueryString(), CoreMatchers.containsString("( 20 - ( 2 * 5 ) )"));
```

Comment:
20-2*5 == 20-(2*5)

Could you instead test for (20-2)*5 since this is different from 20-2*5?

Reply:
Sure. Added the same and pushed it. Also, shall we add all these test cases to the file Section6Test.java only or should there be a separate file for these test cases? I couldn't understand how the Test cases are divided into the various SectionTest files.

Reply:
As I stated earlier. There are actually almost no tests for the SparqlBuilder (ie. no tests will fail if I change random things in the code). I think adding the tests like you have done here should be fine. But @jeenbroekstra should take a look to see if he agrees. Jeen is the project manager for RDF4J.

Reply:
Also, thank you @abhishekag03!

Reply:
Thank you for the valuable suggestions and prompt responses @hmottestad. 
Will wait for @jeenbroekstra to have a look at the changes made so that this issue can be closed.

Reply:
I added a comment on the PR about this, but just to repeat here: I suggest you move these tests out of these `SectionTest` classes completely, and instead introduce a test class `org.eclipse.rdf4j.sparqlbuilder.core.SparqlBuilderTest` where you can add these cases. I'd prefer not to edit the code in the package `org.eclipse.rdf4j.sparqlbuilder.examples` - they are meant to be examples rather than actual code (I don't know if we want to keep them that way, but for now I'd like our actual tests to be separate from it).